Sources/ResourceLocation/ResourceLocation.swift

```diff
@@ -26,20 +26,34 @@ public enum ResourceLocation: Hashable, CustomStringConvertible, Codable {
     
     public var url: URL? {
         switch self {
-        case .remoteUrl(let url):
+        case .remoteUrl(let url, _):
             return url
         case .localFilePath:
             return nil
         }
     }
     
-    public static func from(_ string: String) throws -> ResourceLocation {
+    public var headers: [String: String]? {
+        switch self {
+        case .remoteUrl(_, let headers):
+            return headers
+        case .localFilePath:
+            return nil
+        }
+    }
+    
+    private enum CodingKeys: CodingKey {
+        case url
+        case headers
+    }
+    
+    public static func from(_ string: String, headers: [String: String] = [:]) throws -> ResourceLocation {
```

Comment:
I will think about how we can make this function a bit cleaner after you merge your changes. For now let's leave it as is.